src/dtx/dtx_common.c

```diff
@@ -663,3 +656,211 @@ dtx_handle_resend(daos_handle_t coh, daos_unit_oid_t *oid,
 		return rc >= 0 ? -DER_INVAL : rc;
 	}
 }
+
+static void
+dtx_exec_ops_comp_cb(void **arg)
```

Comment:
why not take a struct dtx_exec_shard_arg ** as input?

Reply:
oh, this is callback called from ABT, it has to be void * here.